ocaml/xapi/xapi_cluster_host.ml

```diff
@@ -53,25 +53,58 @@ let sync_required ~__context ~host =
     end
   | _ -> raise Api_errors.(Server_error (internal_error, ["Cannot have more than one Cluster object per pool currently"]))
 
+let call_api_function_with_alert ~__context ~msg ~cls ~obj_uuid ~body ~(api_func : (Rpc.call -> Rpc.response) -> API.ref_session -> 'a) =
+  Helpers.call_api_functions ~__context
+    (fun rpc session_id ->
+       try
+         api_func rpc session_id |> ignore
+       with err ->
+         error "%s" ExnHelper.(string_of_exn err);
+         Xapi_alert.add ~msg ~cls ~obj_uuid ~body;
```

Comment:
Would be good if the body included the error message too, maybe concatenate it? `body ^ ": " ^ Exnhelper...)

Reply:
Should we keep the error then?

Reply:
No, I think it'll get logged by another exception handler up the call stack.